Sources/TSCBasic/FileSystem.swift

```diff
@@ -224,41 +234,78 @@ public extension FileSystem {
     func getFileInfo(_ path: AbsolutePath) throws -> FileInfo {
         fatalError("This file system currently doesn't support this method")
     }
+
+    func resolveUnicode(_ path: AbsolutePath) -> AbsolutePath {
+        return path
+    }
 }
 
 /// Concrete FileSystem implementation which communicates with the local file system.
 private class LocalFileSystem: FileSystem {
 
+    func resolveUnicode(_ path: AbsolutePath) -> AbsolutePath {
+        #if os(macOS)
+        // The macOS file system enforces NFD.
+        // It can find everything without any help.
+        return path
```

Comment:
Is this contingent on APFS?

Reply:
I have no idea. File systems largely aren’t my area of expertise. Based on [this note on the Wikipedia article for APFS](https://en.wikipedia.org/wiki/Apple_File_System#macOS), I suspect you are at least generally right that my `#if os(macOS)` shortcut here uses the wrong check and not always valid. I have absolutely no idea if there is a more accurate check, or if the shortcut needs to be dropped.